libs/database/src/lib/schemas/base.schema.ts

```diff
@@ -42,3 +43,11 @@ export class CastcleBase {
   @Prop({ type: String })
   queueAction?: CastcleQueueAction;
 }
+
+export class CastcleDocument extends Document {
```

Comment:
Can we extend `CastcleBase` with `Document` instead and remove the types of `<Collection>Document`?